hw_12/hw_12.php

```diff
@@ -0,0 +1,246 @@
+<?php
+
+declare(strict_types=1);
+
+error_reporting(E_ALL);
+ini_set('display_errors', '1');
+
+// Task 1
+
+//function str()
+//{
+//    echo 'Hello, World!';
+//}
+//
+//str();
+
+// Task 2
+
+//echo '<pre>';
+//
+//function isLeapYear($year)
+//{
+//    if ($year % 4 == 0 && $year % 400 == 0) {
+//        echo 'Год високосный';
+//    } else {
+//        echo 'Год не високосный';
+//    }
+//}
+//
+//isLeapYear(2001);
+//
+//echo '</pre>';
+
+// Task 3
+
+//function abbreviation($longstr)
+//{
+//    $words = explode(" ", $longstr);
+//    $abbr = "";
+//
+//    foreach ($words as $word) {
+//        $abbr .= $word[0];
+//    }
+//
+//    return mb_strtoupper($abbr);
+//}
+//
+//echo abbreviation('Cascading Style Sheets');
+
+// Task 4
+
+//function reverse($string)
+//{
+//    $string = mb_str_split($string);
+//    $string = array_reverse($string);
+//    $string = implode('', $string);
+//    echo $string;
+//}
+//
+//echo reverse('Привет, мир!');
+
+// Task 5
+
+//function simpleNum(int $num): bool
+//{
+//    if ($num < 1) {
+//        return false;
+//    };
+//
+//    $i = 2;
+//    while ($i * $i <= $num) {
+//        $i++;
+//        if ($num % $i == 0) {
+//            return false;
+//        }
+//    }
+//    return true;
+//}
+//
+//var_dump(simpleNum(15));
+
+// Task 6
+
+//function countLastWord($str)
+//{
+//    $str = explode(' ', $str);
+//    $lastWord = array_pop($str);
+//    $symbols = ['!', ',', ':', '.', ';'];
+//    $lastWord = str_replace($symbols, '', $lastWord);
+//    $lastWord = mb_strlen($lastWord);
+//    return $lastWord;
+//}
+//
+//echo countLastWord('hello, world!');
+
+//Task 7
+
+//echo '<pre>';
+//
+//function maxNumber($arrNum)
+//{
+//    if (count($arrNum) <= 1) {
+//        return $arrNum;
+//    }
+//
+//    $new_combinations = [];
+//
+//    foreach ($arrNum as $key => $item) {
+//        foreach (maxNumber(array_diff_key($arrNum, array($key => $item))) as $newArrNum) {
+//            $new_combinations[] = $item . $newArrNum;
+//        }
+//    }
+//    return $new_combinations;
+//}
+//
+//$arrNum = [3, 24, 4];
+//echo max(maxNumber($arrNum));
+
+echo '</pre>';
+
+// Task 8
+
+//function isAnagram(string $str1, $str2): bool
+//{
+//    if (count_chars($str1, 1) == count_chars($str2, 1)) {
+//        return true;
+//    } else {
+//        return false;
+//    }
```

Comment:
if лишний
операция сравнения и так вернет bool
```suggestion
//    return count_chars($str1, 1) == count_chars($str2, 1) ;
```